scipy/optimize/cython_optimize/__init__.py

```diff
@@ -0,0 +1,630 @@
+"""
+Cython Optimize Zeros API
+=========================
+This package contains a Cython API for the root finding scalar functions in
+:py:mod:`scipy.optimize` to obtain efficient pure C implementations, although
+faster execution is not guaranteed. For example the Cython Optimize Zeros API
+could be used for native C looping or with Cython ``prange`` for efficient
+execution of large arrays of inputs with the same callbacks. The Cython
+Optimize Zeros API can be imported into Cython code using the following::
+
+    cimport scipy.optimize.cython_optimize as coz
+
+Callback Signatures
+-------------------
+The Cython Optimize Zeros API provides three callback signagures for the root
+finders, two which are safe without the global interpreter lock (GIL) and one
+that uses Python tuples. The functions are grouped by callback signatures into
+Cython modules that can be imported using ``cimport``.
+
+.. py:module:: scipy.optimize.cython_optimize.zeros_struct
+
+``cython_optimize.zeros_struct``
```

Comment:
none of these submodules should be in the public API, there should be only `cython_optimize` with as few functions as possible in it.